pkg/maps/srv6map/policy.go

```diff
@@ -182,3 +129,71 @@ func (m srv6PolicyMap) IterateWithCallback6(cb SRv6PolicyIterateCallback) error
 			cb(&key, value)
 		})
 }
+
+func newPolicyMaps(dc *option.DaemonConfig, lc cell.Lifecycle) (bpf.MapOut[*PolicyMap4], bpf.MapOut[*PolicyMap6], defines.NodeOut) {
+	if !dc.EnableSRv6 {
+		return bpf.MapOut[*PolicyMap4]{}, bpf.MapOut[*PolicyMap6]{}, defines.NodeOut{}
+	}
+
+	m4 := ebpf.NewMap(&ebpf.MapSpec{
+		Name:       policyMapName4,
+		Type:       ebpf.LPMTrie,
+		KeySize:    uint32(unsafe.Sizeof(PolicyKey4{})),
+		ValueSize:  uint32(unsafe.Sizeof(PolicyValue{})),
+		MaxEntries: uint32(maxPolicyEntries),
+		Flags:      unix.BPF_F_NO_PREALLOC,
+		Pinning:    ebpf.PinByName,
+	})
+
+	m6 := ebpf.NewMap(&ebpf.MapSpec{
+		Name:       policyMapName6,
+		Type:       ebpf.LPMTrie,
+		KeySize:    uint32(unsafe.Sizeof(PolicyKey6{})),
+		ValueSize:  uint32(unsafe.Sizeof(PolicyValue{})),
+		MaxEntries: uint32(maxPolicyEntries),
+		Flags:      unix.BPF_F_NO_PREALLOC,
+		Pinning:    ebpf.PinByName,
+	})
+
+	lc.Append(cell.Hook{
+		OnStart: func(ctx cell.HookContext) error {
+			if err := m4.OpenOrCreate(); err != nil {
+				return err
+			}
+			if err := m6.OpenOrCreate(); err != nil {
+				return err
+			}
+			return nil
+		},
+		OnStop: func(ctx cell.HookContext) error {
+			m4.Close()
+			m6.Close()
+			return nil
+		},
+	})
+
+	nodeOut := defines.NodeOut{
+		NodeDefines: defines.Map{
+			"SRV6_POLICY_MAP4":     policyMapName4,
+			"SRV6_POLICY_MAP6":     policyMapName6,
+			"SRV6_POLICY_MAP_SIZE": strconv.FormatUint(maxPolicyEntries, 10),
+		},
+	}
+
+	return bpf.NewMapOut(&PolicyMap4{m4}), bpf.NewMapOut(&PolicyMap6{m6}), nodeOut
+}
+
+// OpenPolicyMaps opens the SRv6 policy maps on bpffs
+func OpenPolicyMaps() (*PolicyMap4, *PolicyMap4, error) {
+	m4, err := ebpf.LoadRegisterMap(policyMapName4)
```

Comment:
Nit: I know this isn't advertised anywhere prominently, but the internal ebpf package is deprecated. Unless there's a feature missing from package bpf, try to use that one instead. Sorry!

Reply:
Oops, ok let me take this opportunity and fix that.

Reply:
Reflected the suggestion.